plugins/services/src/js/utils/ServiceUtil.js

```diff
@@ -44,13 +47,30 @@ const getFindPropertiesRecursive = function(service, item) {
   }, {});
 };
 
+const getFrameworkImages = function(packageName) {
+  const images = Object.assign(
+    {},
+    CosmosPackagesStore.getPackageImages()[packageName]
+  );
+
+  Object.keys(images).forEach(image => {
+    const queryString = images[image].substring(images[image].lastIndexOf("?"));
+    images[image] = RouterUtil.getQueryStringInUrl(queryString).url;
+  });
+
+  return images;
+};
+
 const ServiceUtil = {
   createServiceFromResponse(data) {
     if (ServiceValidatorUtil.isPodResponse(data)) {
       return new Pod(data);
     }
 
     if (ServiceValidatorUtil.isFrameworkResponse(data)) {
+      const packageName = data.labels.DCOS_PACKAGE_NAME;
+      data.images = getFrameworkImages(packageName);
```

Comment:
What happens in the system when the image object is empty?

Reply:
The images are passed to the `FrameworkUtil.getServiceImages()` method, so will return default image if object is empty.